src/github_runner_manager/manager/runner_manager.py

```diff
@@ -341,15 +341,22 @@ def _issue_runner_metrics(self, metrics: Iterator[RunnerMetrics]) -> IssuedMetri
         total_stats: IssuedMetricEventsStats = {}
 
         for extracted_metrics in metrics:
-            try:
-                job_metrics = github_metrics.job(
-                    github_client=self._github.github,
-                    pre_job_metrics=extracted_metrics.pre_job,
-                    runner_name=extracted_metrics.runner_name,
-                )
-            except GithubMetricsError:
-                logger.exception(
-                    "Failed to calculate job metrics for %s", extracted_metrics.runner_name
+            if extracted_metrics.pre_job:
```

Comment:
We can now theoretically have the case where only the runner-installed metrics are extracted and no prejob. So we need this guard.

Reply:
Would it be helpful to have a comment explaining the case in which this happens? I think if we do go for another refactor, without the reasoning being part of the code, I might cause a regression :joy:

Reply:
Have added a comment, thanks.